package/ui/controllers/file_selection.py

```diff
@@ -0,0 +1,154 @@
+from typing import Optional
+import os
+import argparse
+
+from .file_location_persistence import FileLocationPersistence
+from .file_dialog import FileDialogController
+from ui.utils import run_in_workbench
+
+class FileSpec:
+    """ Input/output file specification for a package. """
+    def __init__(self,
+                 arg_name: str,
+                 description: str,
+                 is_output: bool = False,
+                 file_type: Optional[str] = None):
+        """
+        Constructor
+        @In, arg_name, str, optional, the argument flag for the file
+        @In, description, str, the description of the file
+        @In, is_output, bool, optional, whether the file is an output file
+        @In, file_type, str, optional, the type of file
+        @Out, None
+        """
+        self.arg_name = arg_name
+        self.description = description
+        self.is_output = is_output
+        self.file_type = file_type
+
+    def add_to_parser(self, parser: argparse.ArgumentParser) -> argparse.ArgumentParser:
+        """
+        Adds the file specification to the parser.
+        @In, parser, argparse.ArgumentParser, the parser
+        @Out, parser, argparse.ArgumentParser, the parser with the file specification added
+        """
+        # if self.arg_name.startswith('-'):
+        #     parser.add_argument(self.arg_name, nargs=1, required=False, help=self.description)
+        # else:  # positional argument
+        #     parser.add_argument(self.arg_name, nargs='?', help=self.description)
+        parser.add_argument(self.arg_name, nargs='?', help=self.description)
+        return parser
+
+
+class FileSelectionController:
+    """ Controller for the file selection widget. """
+    # Specification for the files that are required for each package. These are used to tie any files
+    # passed from the command line to the file selection widget and helps define what file extensions
+    # are allowed for each file.
+    _file_selection_specs = {
+        'teal': [FileSpec('-iXML', 'TEAL XML File', file_type='xml'),
+                 FileSpec('-iINP','Variable Inputs File', file_type='txt'),
+                 FileSpec('-o', 'Output File', is_output=True)],
+        'ravenframework': [FileSpec('filename', 'RAVEN XML File', file_type='xml')],
+        'heron': [FileSpec('filename', 'HERON Input File', file_type='xml')]
```

Comment:
I agree that we should not. The simple launcher GUI that this snippet is a part of runs HERON/RAVEN/TEAL files directly so they need to be the XML files. Allowing users to specify a .heron input file would require us to open it in Workbench, which may or may not be installed. I think it would be best to keep these two workflows (launcher GUI and Workbench) separate.